lib/std/static_string_map.zig

```diff
@@ -1,446 +1,170 @@
-            len: u32,
-        };
-        const empty_kvs = KVs{
-            .keys = &empty_keys,
-            .values = &empty_vals,
-            .len = 0,
-        };
-        const empty_len_indexes = [0]u32{};
-        const empty_keys = [0][]const u8{};
-        const empty_vals = [0]V{};
-
-        /// Returns a map backed by static, comptime allocated memory.
-        ///
-        /// `kvs_list` must be either a list of `struct { []const u8, V }`
-        /// (key-value pair) tuples, or a list of `struct { []const u8 }`
-        /// (only keys) tuples if `V` is `void`.
-        pub inline fn initComptime(comptime kvs_list: anytype) Self {
-            comptime {
-                var self = Self{};
-                if (kvs_list.len == 0)
-                    return self;
-
-                // Since the KVs are sorted, a linearly-growing bound will never
-                // be sufficient for extreme cases. So we grow proportional to
-                // N*log2(N).
-                @setEvalBranchQuota(10 * kvs_list.len * std.math.log2_int_ceil(usize, kvs_list.len));
+        kvs: []const Kv,
 
-                var sorted_keys: [kvs_list.len][]const u8 = undefined;
-                var sorted_vals: [kvs_list.len]V = undefined;
-
-                self.initSortedKVs(kvs_list, &sorted_keys, &sorted_vals);
-                const final_keys = sorted_keys;
-                const final_vals = sorted_vals;
-                self.kvs = &.{
-                    .keys = &final_keys,
-                    .values = &final_vals,
-                    .len = @intCast(kvs_list.len),
-                };
-
-                var len_indexes: [self.max_len + 1]u32 = undefined;
-                self.initLenIndexes(&len_indexes);
-                const final_len_indexes = len_indexes;
-                self.len_indexes = &final_len_indexes;
-                self.len_indexes_len = @intCast(len_indexes.len);
-                return self;
-            }
-        }
-
-        /// Returns a map backed by memory allocated with `allocator`.
-        ///
-        /// Handles `kvs_list` the same way as `initComptime()`.
-        pub fn init(kvs_list: anytype, allocator: mem.Allocator) !Self {
-            var self = Self{};
-            if (kvs_list.len == 0)
-                return self;
-
-            const sorted_keys = try allocator.alloc([]const u8, kvs_list.len);
-            errdefer allocator.free(sorted_keys);
-            const sorted_vals = try allocator.alloc(V, kvs_list.len);
-            errdefer allocator.free(sorted_vals);
-            const kvs = try allocator.create(KVs);
-            errdefer allocator.destroy(kvs);
-
-            self.initSortedKVs(kvs_list, sorted_keys, sorted_vals);
-            kvs.* = .{
-                .keys = sorted_keys.ptr,
-                .values = sorted_vals.ptr,
-                .len = @intCast(kvs_list.len),
+        /// Initializes the map at comptime
+        pub fn initComptime(comptime kvs_list: anytype) Self {
```

Comment:
This should be `inline fn` so that users don't need to prefix calls with `comptime`

Reply:
What functions exactly do you suggest should be changed? Marking initComptime as inline doesn't seem to remove the need for comptime.
![image](https://github.com/user-attachments/assets/6910f8a4-81bf-4900-8649-c0a8813e6097)

Reply:
you might need to also wrap the body in a comptime block

Reply:
doesn't a parameter being comptime already make the result eagerly evaluated?

Reply:
well yes, but for some reason if you assigned the result to a `const` in a function or test, it would not be treated as comptime-known.